src/Type/Type.php

```diff
@@ -195,6 +196,8 @@ public function isVoid(): TrinaryLogic;
 
 	public function isScalar(): TrinaryLogic;
 
+	public function looseCompare(Type $type, PhpVersion $phpVersion): BooleanType;
```

Comment:
in case we agree on this signature, I would move more logic from `InitializerExprTypeResolver->resolveEqualType` into the type-system